eth/embobj/plus/comm-v2/icub/EoManagement.c

```diff
@@ -156,6 +176,47 @@ extern eOmn_serv_type_t eomn_string2servicetype(const char * string)
     return(eomn_serv_UNKNOWN);    
 }
 
+extern const char * eomn_servicecategory2string(eOmn_serv_category_t servicecat)
+{
+    const char * ret = eomn_serv_category_unknown;
```

Comment:
It is clearly a typo to init the pointer in this way. in any case it does not compile.

have a look at a similar function:

```C++
extern const char * eomn_servicediagnmode2string(eOmn_serv_diagn_mode_t diagnmode)
{
    const char * ret = s_mn_servicediagnmode_strings[eomn_serv_diagn_mode_UNKNOWN];
```

Reply:
sure